Update to samba 4.10.6

chincheta0815 opened the pull request:
This PR is work in progress. So it is not squashed yet.

Could somebody help me with building samba against python.3.5? I do not know how to get that right.

After that's done I would try to get rid of legacy parameters etc. that are not necessary anymore.

chincheta0815 commented:
@pyhalov I saw that you already worked on the samba package. Is there any reason why one shoudl compile samba 32bit and winbind 64bit? 

As of samba-4.10.0 they are deprecating support for python 2.7. I am right now compiling against python 3.5. As this version only is available for 64bit on OI I am thinking of compiling samba 64bit only, dropping all that complex stuff in the Makefile. Otherwise I think python has to be compiled 32bit and 64bit.

What is your opinion?

pyhalov commented:
Really, no reasons. They should be 64-bit both.

chincheta0815 commented:
Made the first compilation working and updated sample-manifest. Missing step: update p5m...

chincheta0815 commented:
@pyhalov I got it compiling, updated the manifest. Now I am a little bit confused on the p5m files... Could you help me a little bit or tell me someone who can?
After that there also needs to be done a final check...

pyhalov commented:
What exact issues do you have with p5m files? There's "Packaging and Delivering Software With
the Image Packaging System in Oracle® Solaris 11.1" guide. It explains IPS packaging. For samba p5m file is mostly auto-generated. With current component you can do "gmake sample-manifest", look at git diff for manifests/sample-manifest.p5m and apply necessary changes to samba.proto-objects. Or just do " gmake sample-manifest METADATA_TEMPLATE=" to generate samba.proto-objects as noted in samba.p5m file. If you regenerate samba.proto-objects, be careful about "dir" actions, our sample-manifest rule doesn't generate them (unlike solaris-userland rule).

chincheta0815 commented:
@pyhalov I have to be more precise. Talking about p5m it's not a problem on finding and reading the docs. There are things like the "dir" actions which are special and cannot be found very easy. But now I know that.

As I state before I threw out all the 32bit stuff. This led to changes in the p5m files and some adaptions are needed. For working/updating the whole show some background info about why all that was created this way would be fine:

- "Makefile" use more in-built functions with less customization I think, e.g. set BITS=64 can maybe avoided by using BUILD_64?
- Is there still a mediation to samba3 needed?
- Are the lines concerned with "SUNWsmbar" still needed?
- How can I see if there are strange attribs or something (samba.pm.common)

It''s mostly not about what a line does (can be read in a doc), but why it's needed (experience).

pyhalov commented:
- BUILD_64 sets BITS=64 for $(BUILD_DIR_64)/.configured target when configure.mk is used, however samba component doesn't use configure.mk currently (look at make-rules/configure.mk). 
- I don't think mediation to samba 3 is needed.
- We usually don't preserve legacy package names.

Let's look at samba.p5m.common:

1) We don't want to provide standard directories in samba package, like dir=etc. Likely we don't want to have dir actions in proto-objects file at all, just add necessary directories to samba.p5m.common. 
Or another option - merge them all (p5m files), like it's done for other packages.
2) Second part of rules is just "pacify pkgdepend/pkglint"
3) Then we set pkg.depend.runpath, as it doesn't necessary can detect it (need to look at sources, how it's done in samba python files themself)
4) Don't know anything about "Private API libgssapi_krb5.so.1 is not yet available in lint cache" and later lines.
5) Mediators for samba 4/3 - likely not needed
6) Service import part likely not needed (we have it by default)
7) Necessary dirs should be preserved.

chincheta0815 commented:
@pyhalov Thanks. That helps a lot. I will then stay with the overall Makefile as is until there might be a "waf" implementation. The point with simply merge all the p5m's seems interisting. This would ease up things a lot.

chincheta0815 commented:
@pyhalov Would you mind helping me in a point: After I got to "gmake publish", there rises now an issue as follows which I do not know how to overcome:

`Couldn't find 'usr/lib/python3.5/vendor-packages/samba/samba/tests/__pycache__/lsa_string.cpython-35.pyc' in any of the specified search directories:
        /root/github.com/oi-userland.git/components/network/samba
        /root/github.com/oi-userland.git/components/network/samba
        /root/github.com/oi-userland.git/components/network/samba/build
        /root/github.com/oi-userland.git/components/network/samba/build/prototype/i386
        /root/github.com/oi-userland.git/components/network/samba/build/prototype/i386/mangled
        /root/github.com/oi-userland.git/components/network/samba/samba-4.10.0
`

Can you give me a hint?
My last commit is squashed for the in the PR.
Thanks in advance.

pyhalov commented:
Packaging system searches for compiled python sources and doesn't find them (see https://wiki.openindiana.org/oi/Building+and+packaging+FAQ).

chincheta0815 commented:
Okay, the transform will then ignore __pycache__ (which for some reason is not there at all after the install into build). But I saw __pycache__ dirs in OI's python3.5 __vendor__ which is only writeable to root. Wouldn't there be problems when python tries to create pyc and __pycache__ files for the first time, because they were not delivered?

> Am 23.03.2019 um 06:45 schrieb Alexander Pyhalov <notifications@github.com>:
> 
> Packaging system searches for compiled python sources and doesn't find them (see https://wiki.openindiana.org/oi/Building+and+packaging+FAQ).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

pyhalov commented:
Likely it will not be a problem. Or you can run python -m compileall , look for example at https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/python/pycairo/Makefile#L80

chincheta0815 commented:
Okay, now I think I understand…
Thanks.

> Am 23.03.2019 um 09:26 schrieb Alexander Pyhalov <notifications@github.com>:
> 
> Likely it will not be a problem. Or you can run python -m compileall , look for example at https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/python/pycairo/Makefile#L80 <https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/python/pycairo/Makefile#L80>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/OpenIndiana/oi-userland/pull/4906#issuecomment-475851102>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACKOnUdj6dZtSkDJq6c6NfNttBV1oXpoks5vZeVAgaJpZM4cBPBp>.
>

chincheta0815 commented:
@pyhalov Well, I followed your hint, but it seems that I am too dump... I changed everything to make use of waf.mk, then I got it compile, made my sample-manifest, added your transform line and now I get a different error message:

`ub.com/oi-userland.git/components/network/samba /root/github.com/oi-userland.git/components/network/samba/build/manifest-i386-samba.mangled >/root/github.com/oi-userland.git/components/network/samba/build/manifest-i386-samba.depend
Couldn't find 'usr/lib/python3.5/vendor-packages/samba/xattr_tdb.cpython-35m.cpython-35m.so' in any of the specified search directories:
        /root/github.com/oi-userland.git/components/network/samba
        /root/github.com/oi-userland.git/components/network/samba
        /root/github.com/oi-userland.git/components/network/samba/build
        /root/github.com/oi-userland.git/components/network/samba/build/prototype/i386
        /root/github.com/oi-userland.git/components/network/samba/build/prototype/i386/mangled
        /root/github.com/oi-userland.git/components/network/samba/samba-4.10.0
Couldn't find 'usr/lib/python3.5/vendor-packages/talloc.cpython-35m.cpython-35m.so' in any of the specified search directories:
        /root/github.com/oi-userland.git/components/network/samba
        /root/github.com/oi-userland.git/components/network/samba
`

Well, not very differnt, just about other files... I also added paths to PKG_PROTO_DIRS, but since that did not help I reverted that change.

Could you please help me? I have to clue how to solve that.
Thanks in advance.

BTW: I had to add "CPP" and "CPPFLAGS" to CONFIGURE_ENV. Shouldn't that be set in waf.mk? RIght now it is not.

Mno-hime commented:
> Couldn't find 'usr/lib/python3.5/vendor-packages/talloc.cpython-35m.cpython-35m.so' in any of the specified search directories: 

I've seen simmilar problem with `waf.mk`: https://github.com/OpenIndiana/oi-userland/pull/4680/files#diff-2c9cf86b220def5636a28547031d234dR75. Maybe just change it to `talloc.so`?

chincheta0815 commented:
> I've seen simmilar problem with `waf.mk`: https://github.com/OpenIndiana/oi-userland/pull/4680/files#diff-2c9cf86b220def5636a28547031d234dR75. Maybe just change it to `talloc.so`?

It seems to be something connected to transform....
The thing is: The files exists where the error message says there is no..

chincheta0815 commented:
Apologies, I have to correct me: The transform rule seems to add cypthon.so, doubling it.
Example:

talloc.cpython-35m.so -> talloc.cpython-35m.cpython-35m.so




> Am 23.03.2019 um 23:49 schrieb Michal Nowak <notifications@github.com>:
> 
> Couldn't find 'usr/lib/python3.5/vendor-packages/talloc.cpython-35m.cpython-35m.so' in any of the specified search directories:
> 
> I've seen simmilar problem with waf.mk: https://github.com/OpenIndiana/oi-userland/pull/4680/files#diff-2c9cf86b220def5636a28547031d234dR75. Maybe just change it to talloc.so?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

Mno-hime commented:
> Apologies, I have to correct me: The transform rule seems to add cypthon.so, doubling it. Example: talloc.cpython-35m.so -> talloc.cpython-35m.cpython-35m.so

Yes, that's what's happening, it's a build system problem. The link I posted is a workaround for immediate consumption, should you be interrested in such a thing for now.

chincheta0815 commented:
I actually git it:

Using 

PY3_CYTHON_NAMING =\#

Did not show that message anymore.


> Am 24.03.2019 um 09:49 schrieb Michal Nowak <notifications@github.com>:
> 
> Apologies, I have to correct me: The transform rule seems to add cypthon.so, doubling it. Example: talloc.cpython-35m.so -> talloc.cpython-35m.cpython-35m.so
> 
> Yes, that's what's happening, it's a build system problem. The link I posted is a workaround for immediate consumption, should you be interrested in such a thing for now.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

chincheta0815 commented:
@pyhalov and @Mno-hime I finally got a package. I had some trouble with adapting the transforms in the context of python.

Here's what I changed:

- changed to samba-4.10.0,
- build samba 64bit only,
- changed to python3.5 as samba has deprecated python2.7 support,
- changed to using waf.mk,
- removed mediator links,
- removed old SUNsmb* packages,
- tried to clean up from seemingly unnecessary parts/transforms, etc.

I am now testing a little bit.

Would you give me some input on the work, where there are issues and how to improve?

chincheta0815 commented:
@pyhalov @Mno-hime I have an issue with an error message coming from tdb_mutex_init() which deals with pthreads that causes a sever bug that I am trying hard on to overcome. Since I am not a computer engineer I would appreciate some help...

Samba-mailing list told me to add debug statements to the code...

Is there any known difference that has to be kept in mind when compiling samba/threads? pkgsrc of SmartOS does not patch their samba-4.10 code...

Or is there a known issue testing samba in a virtualbox as I do?

Maybe I did some mistakes while packaging that resolve to the bug I have?

I really appreciate any help on this issue as I like to get that show in the road.

Here is the error message:
`[2019/04/01 09:51:41.112485, 0] ../../lib/tdb_wrap/tdb_wrap.c:64(tdb_wrap_log)
tdb(/var/samba/lock/gencache.tdb): tdb_open_ex: tdb_mutex_init failed for /var/samba/lock/gencache.tdb: Device busy
[2019/04/01 09:51:41.141396, 0] ../../lib/util/become_daemon.c:136(daemon_ready)
daemon_ready: daemon 'smbd' finished starting up and ready to serve connections`

Here truss output:
`2116:   setgroups(11, 0x011FCC40)                       = 0
2116:   setregid(4294967295, 0)                         = 0
2116:   getgid()                                        = 0 [0]
2116:   setreuid(4294967295, 0)                         = 0
2116:   getuid()                                        = 0 [0]
2116:   umask(0)                                        = 0
2116:   mkdir("/var/samba/lock", 0755)                  Err#17 EEXIST
2116:   umask(0)                                        = 0
2116:   lstat("/var/samba/lock", 0xFFFFFD7FFFDFB240)    = 0
2116:   open("/var/samba/lock/gencache.tdb", O_RDWR|O_CREAT, 0644) = 21
2116:   fcntl(21, F_GETFD, 0x00000000)                  = 0
2116:   fcntl(21, F_SETFD, 0x00000001)                  = 0
2116:   fcntl(21, F_SETLKW, 0xFFFFFD7FFFDFB040)         = 0
2116:   read(21, " T D B   f i l e\n\0\0\0".., 168)     = 168
2116:   fstat(21, 0xFFFFFD7FFFDFB130)                   = 0
2116:   fcntl(21, F_SETLK, 0xFFFFFD7FFFDFB040)          = 0
2116:   mmap(0x00000000, 241664, PROT_READ|PROT_WRITE, MAP_SHARED, 21, 0) = 0xFFFFFD7FED194000
2116:   lwp_mutex_register(0xFFFFFD7FED1940C8, 0xFFFFFD7FED2A30D0) = 0
`

And here the mailing list conversation:
https://lists.samba.org/archive/samba/2019-April/222158.html

chincheta0815 commented:
I am totally lost concerning this pull request...
Joyent seem to get it working without any problem (https://github.com/joyent/pkgsrc/blob/trunk/net/samba4/Makefile).
The samba-devs thought of an issue in Illumos pthread support...
What am I doing wrong???

pyhalov commented:
I've seen something similar when /dev/epoll device was not available in zone.

pyhalov commented:
Do you trace all samba processes?

chincheta0815 commented:
> Do you trace all samba processes?

I do not have that much experience in using truss... I did: "truss -f smbd > truss.out 2>&1"

Diggin deeper in it with some errno I saw that "ret = pthread_mutexattr_init(&ma)" showed errno 16.
(file is lib/tdb/common/mutex.c line 570)

chincheta0815 commented:
An addition: samba-4.9.5 does not show that issue.

pyhalov commented:
samba.proto-objects is incorrect.
a) it contains references to usr/include/samba-4.10
b) it contains pkg.fmri and other attributes, which should be present in samba.p5m
c) it doesn't contain 'dir' entries.

To create it, you can use something like 
```
/usr/bin/pkgsend generate  /export/home/alp/srcs/oi-userland/components/network/samba-4.10/build/prototype/i386 | /usr/bin/pkgmogrify -D PERL_ARCH=i86pc-solaris-64int -D PERL_VERSION=5.22 -D PG_VERSION=9.4 -D PG_VERNUM=94 -D PG_BASEPKG=database/postgres-94 -D MYSQL_VERSION=10.1 -D MYSQL_VERNUM=101 -D MYSQL_BASEPKG=database/mariadb-101 -D IPS_COMPONENT_RE_VERSION=4\\.10\\.0 -D COMPONENT_RE_VERSION=4\\.10\\.0 -D PYTHON_2.7_ONLY=# -D PYTHON_3.4_ONLY=# -D PYTHON_3.5_ONLY=# -D PYTHON_32_ONLY= -D COMPONENT_CLASSIFICATION="org.opensolaris.category.2008:" -D PY3_ABI3_NAMING="#" -D PY3_CYTHON_NAMING="#" -D ARC_CASE="" -D TPNO="" -D MACH="i386" -D MACH32="i86" -D MACH64="amd64" -D BUILD_VERSION="5.11-2018.0.0.0" -D OS_VERSION="5.11" -D PKG_SOLARIS_VERSION="5.11" -D CONSOLIDATION="userland" -D COMPONENT_VERSION="4.10.0" -D IPS_COMPONENT_VERSION="4.10.0" -D HUMAN_VERSION="" -D COMPONENT_ARCHIVE_URL="http://us1.samba.org/samba/ftp/stable/samba-4.10.0.tar.gz" -D COMPONENT_PROJECT_URL="http://www.samba.org/" -D COMPONENT_NAME="samba" -D COMPONENT_FMRI="service/network/samba" -D COMPONENT_LICENSE_FILE="samba.license" -D COMPONENT_SUMMARY="samba - A Windows SMB/CIFS fileserver for UNIX" -D COMPONENT_DESCRIPTION="" -D COMPONENT_LICENSE="GPLv3" -D HG_REPO="" -D HG_REV="" -D HG_URL="" -D COMPONENT_HG_URL="" -D COMPONENT_HG_REV="" -D GIT_COMMIT_ID="" -D GIT_REPO="" -D GIT_TAG="" -D MACH="i386" -D MACH32="i86" -D MACH64="amd64" -D PUBLISHER="userland" -D PUBLISHER_LOCALIZABLE="userland-localizable" -D USERLAND_GIT_REMOTE="https://github.com/pyhalov/oi-userland.git" -D USERLAND_GIT_BRANCH="oi/hipster" -D USERLAND_GIT_REV="4dbe59fa4ae7e38a129b419cf6528faeabb64b22" -I/export/home/alp/srcs/oi-userland/components/network/samba-4.10 /dev/fd/0 /export/home/alp/srcs/oi-userland/transforms/generate-cleanup |         sed -e '/^$/d' -e '/^#.*$/d' -e '/\.la$/d' -e '/\.pyo$/d' -e '/usr\/lib\/python[23]\..*\.pyc$/d'         -e '/.*\/__pycache__\/.*/d'  |         /usr/bin/pkgfmt > samba.proto-objects
```
which is command used by gmake-sample-manifest, modified to preserve directories and avoid adding pkg meta-data from template.

pyhalov commented:
Makefile misses ```COMPONENT_CLASSIFICATION =      System/File System```

chincheta0815 commented:
> 
> 
> samba.proto-objects is incorrect.
> a) it contains references to usr/include/samba-4.10
That's due to changes in the configure opts I actually did once. I saw that this is a change in the paths and reverted this. Maybe I just uploaded an old proto-objects file.

> b) it contains pkg.fmri and other attributes, which should be present in samba.p5m
This is supposed to be a change compared to 4.9.5 as gmake sample-manifest always includes pkg.fmri. sampa.p5m includes (or is intended to include) the proto-objects.

> c) it doesn't contain 'dir' entries.
> 
> To create it, you can use something like
> 
> ```
> /usr/bin/pkgsend generate  /export/home/alp/srcs/oi-userland/components/network/samba-4.10/build/prototype/i386 | /usr/bin/pkgmogrify -D PERL_ARCH=i86pc-solaris-64int -D PERL_VERSION=5.22 -D PG_VERSION=9.4 -D PG_VERNUM=94 -D PG_BASEPKG=database/postgres-94 -D MYSQL_VERSION=10.1 -D MYSQL_VERNUM=101 -D MYSQL_BASEPKG=database/mariadb-101 -D IPS_COMPONENT_RE_VERSION=4\\.10\\.0 -D COMPONENT_RE_VERSION=4\\.10\\.0 -D PYTHON_2.7_ONLY=# -D PYTHON_3.4_ONLY=# -D PYTHON_3.5_ONLY=# -D PYTHON_32_ONLY= -D COMPONENT_CLASSIFICATION="org.opensolaris.category.2008:" -D PY3_ABI3_NAMING="#" -D PY3_CYTHON_NAMING="#" -D ARC_CASE="" -D TPNO="" -D MACH="i386" -D MACH32="i86" -D MACH64="amd64" -D BUILD_VERSION="5.11-2018.0.0.0" -D OS_VERSION="5.11" -D PKG_SOLARIS_VERSION="5.11" -D CONSOLIDATION="userland" -D COMPONENT_VERSION="4.10.0" -D IPS_COMPONENT_VERSION="4.10.0" -D HUMAN_VERSION="" -D COMPONENT_ARCHIVE_URL="http://us1.samba.org/samba/ftp/stable/samba-4.10.0.tar.gz" -D COMPONENT_PROJECT_URL="http://www.samba.org/" -D COMPONENT_NAME="samba" -D COMPONENT_FMRI="service/network/samba" -D COMPONENT_LICENSE_FILE="samba.license" -D COMPONENT_SUMMARY="samba - A Windows SMB/CIFS fileserver for UNIX" -D COMPONENT_DESCRIPTION="" -D COMPONENT_LICENSE="GPLv3" -D HG_REPO="" -D HG_REV="" -D HG_URL="" -D COMPONENT_HG_URL="" -D COMPONENT_HG_REV="" -D GIT_COMMIT_ID="" -D GIT_REPO="" -D GIT_TAG="" -D MACH="i386" -D MACH32="i86" -D MACH64="amd64" -D PUBLISHER="userland" -D PUBLISHER_LOCALIZABLE="userland-localizable" -D USERLAND_GIT_REMOTE="https://github.com/pyhalov/oi-userland.git" -D USERLAND_GIT_BRANCH="oi/hipster" -D USERLAND_GIT_REV="4dbe59fa4ae7e38a129b419cf6528faeabb64b22" -I/export/home/alp/srcs/oi-userland/components/network/samba-4.10 /dev/fd/0 /export/home/alp/srcs/oi-userland/transforms/generate-cleanup |         sed -e '/^$/d' -e '/^#.*$/d' -e '/\.la$/d' -e '/\.pyo$/d' -e '/usr\/lib\/python[23]\..*\.pyc$/d'         -e '/.*\/__pycache__\/.*/d'  |         /usr/bin/pkgfmt > samba.proto-objects
> ```
> 
I do not remember exactly right now, but I think I moved the dirs from the proto-objects to the samba.p5m. This might be incorrect.

> which is command used by gmake-sample-manifest, modified to preserve directories and avoid adding pkg meta-data from template.
I used gmake sample-manifest. Accoring to e.g. lmdb's p5m it was intended by me to include the meta-data. It mght be a little bit confusing as there is some meta-data that was not created automatically be gmake sample-manifest in the samba.p5m and not in the samba.proto-objects.

I changed some things, but these seemed reasonable to me, trying to make the package less complex. Also avoiding the 32bit version...

After all I got a pkg that was installable... Maybe just a not working one?

chincheta0815 commented:
@pyhalov I looked into the files. There was a non-updated sample-manifest, that contained samba-4.0 include-dirs. Sorry for that. That was maybe due to the testing I did. The other changes you mention are included in the samba.p5m and samba.proto-files. Here I simply changed them into a combination of samba.p5m which is hand-made and samba.proto which is gmake sample-manifest made. If I should change that to the samba-4.9.5 version okay, but I would rather do that after the pkg and more the build works.

Right now I get a installable package, that starts sambe but then I get the gencache.tdb device busy issue that I am not able to solve...

Mno-hime commented:
@chincheta0815 Makefile is now in conflict.

Did you ask about `tdb_mutex_init failed` on the illumos devel mailing list (just saw your email on oi-dev) or #illumos IRC channel at least? There happen to be folks who are interested in SMB.

chincheta0815 commented:
> @chincheta0815 Makefile is now in conflict.

Ahh. Okay Have to repair that tonight.

I asked on oi-dev and on samba-technical for now. They told me it should be a problem with pthreads, but this seems weird as samba-4.9.5 compiles on the same system. But if illumos-dev would be an option I could try there.

Commit:
update to samba-4.10.5

chincheta0815 commented:
@Mno-hime Okay the conflict should be solved and I asked in illumos-developer.

chincheta0815 commented:
@pyhalov @Mno-hime I got it working respecting issue 9959 (https://github.com/illumos/illumos-gate/commit/6da2547363ddbd247ee9513db83b05a31bca31af). What puzzles me: The issue seems to be solved in the underlying illlumos code base, but does not memset the mutex before init. Maybe this is concerned with MUTEX_MAGIC? No idea why. Feel free to test and give hints...

Mno-hime commented:
@chincheta0815 Is the PR still WIP? Does 4.10 work fine with the referenced illumos fix present in the OS?

chincheta0815 commented:
@Mno-hime Well, so far I had no error messages. So it seems to work. The thing I am not familiar with is that I mainly re-implemented what the illumos fix by issue #9959 does. The only thing that the path I added does not it checking the MUTEX_MAGIC. This could be harmful. I got some hints in the illumos-dev mailing list that say the samba fix is okay. The originator of the illumos issue #9959 does say that it simply might seem to be okay and does not prrof for functionality. The samba-devs did not care rather than saying it is an OS issue. So I am a littte bit lost... I would say WIP until someone with more knowledge helps. BTW: SmartOS seems to have the same issue due to the illumos-dev list and my correspondence with that list.

Mno-hime commented:
Thanks for the explanation, @chincheta0815. I think, at some point, in case of no progress in assuring that the possible problem isn't actually harmful, and no other issue, we will do a "leap of faith" and merge it. Come back to this PR in couple of weeks and share your thoughts, please. Thanks so far!

chincheta0815 commented:
@Mno-hime Say it this way: From my point of view you can easily merge it right now. For me I am done with testing as I have not seen issues on my system. But I use samba not very excessively with all it's features. So if you want: Merge it.

chincheta0815 commented:
Just renamed the PR.

Mno-hime commented:
Thanks for the explanation @chincheta0815. We will merge this after OpenIndiana snapshot 2019.04 is released (due in few weeks).

Mno-hime commented:
@chincheta0815 Add your copyright claim at the top of the component `Makefile`, please. Otherwise, I reviewed this and since we are now after 2019.04 snapshot, I approve this.

chincheta0815 commented:
@Mno-hime Sorry for the delay. I added the copyright (hopefully everywhere where necessary) and updated to version 4.10.3.

Mno-hime reviewed (approved):
Thank you very much for your contribution!

pyhalov commented on components/network/samba/samba.p5m:
```diff
@@ -19,37 +19,56 @@
-#   gmake sample-manifest METADATA_TEMPLATE=
-# as a "sample-manifest"-file (in build/ dir) and is pkgfmt clean
-# All individual actions and attribute settings on pkg objects should
-# be performed by "<transform ...> (pkgmogrify) actions in "samba.p5m.common".
```
Should we just merge samba.p5m and samba.proto-objects?

chincheta0815 replied:
I would not. The idea was: One simply runs the gmake sample-manifest as it is without any arguments. This creates the file needed with all its auto-generated header info etc. This file is then included into samba.p5m, which has not to be changed. In samba.p5m some hand-made additions are made. 

Otherwise one always has to do a "special arguments command" (which you might forget to do that way) for a repeatedly needed "gmake sample-manifest". That's why I did it this way.

pyhalov replied:
I mean, we can reformat samba.p5m, including samba.proto-objects in it, and treat it as any other component. When you update it, you just look at git diff manifests/sample-manifest.p5m, and update samba.p5m as necessary.

chincheta0815 replied:
Maybe I do not get your idea. (Maybe because it's Monady ;o)

samba.p5m includes samba.proto-objects automatically (see https://github.com/chincheta0815/oi-userland/blob/7e4df109667e92664dc105044d01326d4770c9d7/components/network/samba/samba.p5m#L48). By simply doing a "gmake sample-manifest" plus a "cp ./manifests/sample-manifest.p5m ../samba.proto-objects" the workflow does the same as you want to do, but at least for me, easier, less error prone. (Without the diff and handily checking the files contents, I often forgot to use MANIFEST_TEMPLATE= and caused some issues...)

Maybe I just do not get your thought!?

pyhalov replied:
@Mno-hime What do you think? 
I understand that current approach works fine, it's just different from other components, and this worries me.

chincheta0815 replied:
Short addition: 
Yes, the current approach (everything before 4.10.x works fine; a suer tends to do mistakes with it).

The new files were not intended to be changed. After the working on the files they turned out to be more useful. They just grew by modifying the old files into their new layout. A user (me) often does not find the line stating that "MANIFEST_TEMPLATE=" is needed, ending up with searching what caused and error. And useing that argument is quite special to the samba-pkg.

Mno-hime replied:
I find both ways of building this equally awkward, so I am fine with what the author choose.

pyhalov commented on components/network/samba/samba.p5m:
```diff
@@ -19,37 +19,56 @@
-set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
-set name=org.opensolaris.arc-caseid value=PSARC/2015/008
-set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+# General dirs generated in sample-manifest but with bad attribs.
```
Our sample-manifest doesn't generate dir actions

pyhalov replied:
@chincheta0815 Please, remove dir drop rules from here and let's merge it.

pyhalov commented:
Do we need additional testing for this component?

Mno-hime commented:
> Do we need additional testing for this component?

I have no idea about Samba, nor means to test this (e.g. a Windows to test against). So, I wont test it myself.

chincheta0815 commented:
I mainly orientated myself on the former pkg. I could start it and it worked. There is a vast amount of use cases and optiones, etc. so testing after it is compiled is really a lot of stuff to do. I would not say that I "tested" it tough. I started it and for my very simple purposes it worked.

pyhalov commented:
Looked at my test zone and found core:
```
fffffd7febb8772a _lwp_kill () + a
 fffffd7febb19670 raise (6) + 20
 fffffd7febaf1cd8 abort () + 98
 fffffd7fc9c0968e dump_core () + 12e
 fffffd7fc9c28526 smb_panic_s3 () + 46
 fffffd7fca5db7bf smb_panic () + 1f
 fffffd7fca5dba05 sig_fault () + 65
 fffffd7febb80556 __sighndlr () + 6
 fffffd7febb72f3b call_user_handler (b, 0, fffffd7fffdfec10) + 1db
 fffffd7febb732ae sigacthandler (b, 0, fffffd7fffdfec10) + 10e
 --- called from signal handler with signal 11 (SIGSEGV) ---
 fffffd7fca884e12 smbXsrv_session_create () + 32
 fffffd7fca80ee11 reply_sesssetup_and_X () + 341
 fffffd7fca851ec9 switch_message () + 299
 fffffd7fca854646 process_smb () + 1f6
 fffffd7fca8558be smbd_server_connection_read_handler () + ce
 fffffd7fc97ca204 tevent_common_invoke_fd_handler () + 84
 fffffd7fc97d1657 epoll_event_loop_once () + 277
 fffffd7fc97cf510 std_event_loop_once () + 40
 fffffd7fc97c9855 _tevent_loop_once () + 95
 fffffd7fc97c9ad3 tevent_common_loop_wait () + 23
 fffffd7fc97cf490 std_event_loop_wait () + 40
 fffffd7fca856c3c smbd_process () + 69c
 000000000040e679 smbd_accept_connection () + 1c9
 fffffd7fc97ca204 tevent_common_invoke_fd_handler () + 84
 fffffd7fc97d1657 epoll_event_loop_once () + 277
 fffffd7fc97cf510 std_event_loop_once () + 40
 fffffd7fc97c9855 _tevent_loop_once () + 95
 fffffd7fc97c9ad3 tevent_common_loop_wait () + 23
 fffffd7fc97cf490 std_event_loop_wait () + 40
 0000000000410400 main () + 1b90
 0000000000409233 _start_crt () + 83
 0000000000409198 _start () + 18
```

chincheta0815 commented:
> Looked at my test zone and found core:
> 
> ```
> fffffd7febb8772a _lwp_kill () + a
>  fffffd7febb19670 raise (6) + 20
>  fffffd7febaf1cd8 abort () + 98
>  fffffd7fc9c0968e dump_core () + 12e
>  fffffd7fc9c28526 smb_panic_s3 () + 46
>  fffffd7fca5db7bf smb_panic () + 1f
>  fffffd7fca5dba05 sig_fault () + 65
>  fffffd7febb80556 __sighndlr () + 6
>  fffffd7febb72f3b call_user_handler (b, 0, fffffd7fffdfec10) + 1db
>  fffffd7febb732ae sigacthandler (b, 0, fffffd7fffdfec10) + 10e
>  --- called from signal handler with signal 11 (SIGSEGV) ---
>  fffffd7fca884e12 smbXsrv_session_create () + 32
>  fffffd7fca80ee11 reply_sesssetup_and_X () + 341
>  fffffd7fca851ec9 switch_message () + 299
>  fffffd7fca854646 process_smb () + 1f6
>  fffffd7fca8558be smbd_server_connection_read_handler () + ce
>  fffffd7fc97ca204 tevent_common_invoke_fd_handler () + 84
>  fffffd7fc97d1657 epoll_event_loop_once () + 277
>  fffffd7fc97cf510 std_event_loop_once () + 40
>  fffffd7fc97c9855 _tevent_loop_once () + 95
>  fffffd7fc97c9ad3 tevent_common_loop_wait () + 23
>  fffffd7fc97cf490 std_event_loop_wait () + 40
>  fffffd7fca856c3c smbd_process () + 69c
>  000000000040e679 smbd_accept_connection () + 1c9
>  fffffd7fc97ca204 tevent_common_invoke_fd_handler () + 84
>  fffffd7fc97d1657 epoll_event_loop_once () + 277
>  fffffd7fc97cf510 std_event_loop_once () + 40
>  fffffd7fc97c9855 _tevent_loop_once () + 95
>  fffffd7fc97c9ad3 tevent_common_loop_wait () + 23
>  fffffd7fc97cf490 std_event_loop_wait () + 40
>  0000000000410400 main () + 1b90
>  0000000000409233 _start_crt () + 83
>  0000000000409198 _start () + 18
> ```

Interesting. This error is new!? I have some issue with my virtualbox for testing as there is no working update for MacOS right now. As soon as this works I will update samba to 4.10.4 and try to reproduce the issue...

chincheta0815 commented:
I am back again.

After updating to samba 4.10.5 I got a clean compile and install.
I installed samba using pkg and then it started without a problem

Could you retry and tell me what you did exactly when you receive a coredump?

Commit:
update to samba-4.10.6

chincheta0815 commented:
Update to samba-4.10.6.
@pyhalov Could you please check again? I could not reproduce the error you described. Maybe I need more information about what you did...

pyhalov commented:
I've rechecked. So far it works fine. 
Is there anyone else who want to test it?
@Mno-hime ?

Mno-hime commented:
@pyhalov I can't test it. My approval still stands.
